zuul/connection/github.py

```diff
@@ -425,6 +432,23 @@ def mergePull(self, owner, project, pr_number, commit_message='',
         if not result:
             raise Exception('Pull request was not merged')
 
+    def getCommitStatuses(self, owner, project, sha):
+        # A ref can have more than one status from each context,
+        # however the API returns them in order, newest first.
+        # So we can keep track of which contexts we've already seen
+        # and throw out the rest.
+        repository = self.github.repository(owner, project)
+        commit = repository.commit(sha)
+        seen = []
+        statuses = []
+        for status in commit.statuses():
+            if status.context not in seen:
+                statuses.append("%s:%s" % (status.context, status.state))
+                seen.append(status.context)
+
```

Comment:
I'm not sure it could. You may wind up with `bonnyci:success` and `bonnyci:error` in the list, as they're unique, but they conflict. We'd want only the latest status from the `bonnyci` context. Context is the the unique key.